### DISCUSSION TITLE
rdp session shadowing

### DISCUSSION THREAD

**Discussion by akarl10** [FreeRDP/FreeRDP#6747] [Q&A] [answered]
It is widely believed that with remote desktop you can only create your own session and not share your screen.

http://woshub.com/rds-shadow-how-to-connect-to-a-user-session-in-windows-server-2012-r2/
http://woshub.com/rdp-session-shadow-to-windows-10-user/

I think if this works like described here this would be a great substitute for things like vnc and tools like that.

I did not find any traces how to do this with freerdp, but I think this is simply not implemented, given how unknown this option is.

Does somebody know more about this?

**Comment by akallabeth** [comment, marked as answer]
1. there is `freerdp-shadow` which can shadow an existing session. (currently has either basic authentication of the system, e.g. PAM or none at all)
2. there is remote assistance mode on windows which allows a RDP client to connect to the session of a user (view only or remote control) which also works with freerdp as client

**Reply by digitalsignalperson** [reply]
Per the OP's 2nd link, I was able to shadow one Windows 10 PC to another with 
`mstsc /shadow:1 /v:192.168.0.X /prompt /noConsentPrompt /control` 
where 192.168.0.X is the IP of the system to shadow, which I did the group policy config as instructed to allow full control. 

That is way nicer than the remote assistance method!

Is freerdp capable of connecting in the same way with flags that mstsc would?  ` /shadow:1 /prompt /noConsentPrompt /control`. It would be killer to be able to shadow a windows session from linux.

~~Unclear to me if this issue is the same question https://github.com/FreeRDP/FreeRDP/issues/8063 because the RDP server with shadow capability is already built into windows~~ oh nvm I'm looking for this https://github.com/FreeRDP/FreeRDP/issues/5656

**Reply by My1** [reply]
One difference between renote assistance and rdp shadow might be tho that shadowing works based on a direct connection and likely actually works, unlike remote assistance which i almost never got to work and that was windows to windows.

Also how does remote assistance mode handle UAC. The new quick support for example does not forward UAC last time i checked and therefore is unsuitable for connecting to workstations where the user doesn't have admin but you need to supply creds

**Reply by Jorilx** [reply]
@akallabeth About point 2, I'm sorry, how can I tell freerdp to connect as "Unsolicited" remote assistance?

**Comment by MikeUgra** [comment]
Sorry, didn't understand, freerdp-shadow - is connection setting or what?

**Reply by akallabeth** [reply]
@MikeUgra `freerdp-shadow-cli` is a application of `FreeRDP` that mirrors a running `X11` session when started

**Comment by MikeUgra** [comment]
many thanks!
